ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Round.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.math;
+
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.function.udf.UdfSchemaProvider;
+import io.confluent.ksql.schema.ksql.SqlBaseType;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.KsqlException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.List;
+
+/*
+The rounding behaviour implemented here follows that of java.lang.Math.round()
+It's an implementation of rounding "half up". This means we round to the nearest integer value and
+in the case we are equidistant from the two nearest integers we round UP to the nearest integer.
+This means:
+
+ROUND(1.1) -> 1
+ROUND(1.5) -> 2
+ROUND(1.9) -> 2
+
+ROUND(-1.1) -> -1
+ROUND(-1.5) -> -1    Note this is not -2! We round up and up is in a positive direction.
+ROUND(-1.9) -> -2
+ */
+@UdfDescription(name = "Round", description = Round.DESCRIPTION)
+public class Round {
+
+  static final String DESCRIPTION =
+      "Returns the value rounded to the specified number of decimal places. "
+      + "If the number of decimal places is omitted it defaults to zero."
+      + "Numbers are rounded to the nearest digit at the last required decimal place. "
+      + "If the digit to the right of the last required decimal place is equidistant (i.e. a 5) "
+      + "then it is rounded up.";
+
+  private static final MathContext SIMPLE_ROUND_HALF_UP =
+      new MathContext(1, RoundingMode.HALF_UP);
+
+  @Udf
+  public Long round(@UdfParameter final Double val) {
+    return val == null ? null : Math.round(val);
+  }
+
+  @Udf
+  public Double round(@UdfParameter final Double val, @UdfParameter final Integer decimalPlaces) {
+    // Note that we MUST BigDecimal.valueOf(double) to create the BD to ensure correctness
```

Comment:
I would suggest replacing comments such as these with specific test cases that will fail if this is not done.

Reply:
The test cases already exist. I am not a big fan of too many comments but I think a few pertinent ones can be useful.

Reply:
Given there are tests that will fail if someone changes this code to not use `BigDecimal.valueOf(double)`, I don't see how this comment adds any value.  All code does certain things to ensure its implemented correctly, but we don't add comments everywhere.

My issue with comments such as these is that they invariably end up out of date.  Someone changes the code, (in a way were all tests still pass), but not the comment.   Having a comment here won't stop someone changing the code - it's failing tests that will make them realize there change is not right.

If you feel strongly about it then by all means leave them in. Just don't be surprised is someone else removes them if they're making changes in the file.

Reply:
I will move this to the class header

Reply:
Ok, I'll delete it. But I will be prepared to answer questions "Why the hell are you constructing the BigDecimal in that way?" ;)

Reply:
Done